### PR TITLE
feat: use `charmcraft` structured status output for `release-charm`

### LINKED ISSUE #45: `release-action` fails to parse `charmcraft status`'s output if tracks have branches
### Bug Description

For tracks that have charmhub branches, such as:
```
charmcraft status metacontroller-operator                                
Track    Base                  Channel                    Version    Revision    Expires at                                                                                             
latest   ubuntu 20.04 (amd64)  stable                     2          2                                                                                                                  
                               candidate                  5          5                                                                                                                  
                               beta                       ↑          ↑                                                                                                                  
                               edge                       13         13                                                                                                                 
                               edge/add-charming-actions  9          9           2022-06-30T01:00:00Z                                                                                   
                               edge/resources-removal     12         12          2022-07-09T01:00:00Z                   
```
the `release-charm` action fails due to incorrect parsing of the available channels.  

### To Reproduce

See [this action run](https://github.com/canonical/metacontroller-operator/runs/7073697720?check_suite_focus=true)

### Environment

See [this action run](https://github.com/canonical/metacontroller-operator/runs/7073697720?check_suite_focus=true)

### Relevant log output

```shell
See [this action run](https://github.com/canonical/metacontroller-operator/runs/7073697720?check_suite_focus=true)
```

### Additional context

`charmcraft` edge now allows for structured output using `charmcraft status charmName --format json`.  This would let us avoid parsing the human readable text.  There are some [growing-pains](https://github.com/canonical/charmcraft/issues/795) with this output, however.

### PR DESCRIPTION
**Pull request by ca-scribner**
This is a first draft at using `charmcraft status`'s formatted output for the `release-charm` action in order to handle releasing charms that have charmhub branches (see #45 for details on the issue).

This is not the best javascript, but instead a quick hack to get something working.

fixes #45

### CODE REVIEW

**Comment by ca-scribner** [comment]
fyi [this change](https://github.com/canonical/charmcraft/pull/802) to charmcraft will affect the data you're parsing

**Comment by agathanatasha** [comment]
I think finding safer channels when promoting charms is confusing and counter-intuitive. Promoting charm is a more "dangerous" operation than downloading or deploying charm. It is also an one-off operation. I think it is more important to keep the logic simple and easy to understand when we come back to it every few months. With the upload-charm action, there would be at least one open channel. It is an edge case to attempt promotion from a tracking channel. 
If we do support tracking channel, there isn't an easy way to warn devs when they do promote charm from a tracking channel. The action would finish successfully, but obfuscate the fact that it wasn't run exactly as specified in the action input.

File: release-charm/action.yaml [hunk 1]
```diff
@@ -12,6 +12,21 @@ inputs:
     description: |
       Origin Channel from where the charm that needs to be promoted will be
       pulled unless `revision` is set.
```
Comment (simskij) [inline release-charm/action.yaml:14]:
```suggestion
      pulled.
```

**Review by simskij — APPROVED** [review]

**Comment by simskij** [comment]
> Finding safer channels when promoting charms is confusing and counter-intuitive. Promoting charm is a more "dangerous" operation than downloading or deploying charm. It is also an one-off operation. I think it is more important to keep the logic simple and easy to understand when we come back to it every few months. With the upload-charm action, there would be at least one open channel. It is an edge case to attempt promotion from a tracking channel. If we do support tracking channel, there isn't an easy way to warn devs when they do promote charm from a tracking channel. The action would finish successfully, but obfuscate the fact that it wasn't run exactly as specified in the action input.

I agree with you on this one. It's definitely preferable to fail on all corner cases then to do some magic behind the scenes and leaving the user hoping for the best.

**Pull request merged**